File: CS/ReportingEUDSaveReportWithoutConnectionParams/Form1.cs
using DevExpress.DataAccess.Wizard.Services;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.UserDesigner;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WinEUDSaveReportWithoutConnectionParams {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        private void simpleButton1_Click(object sender, EventArgs e) {
            OpenReportDesigner();
        }
        CustomConnectionStorageService connectionStorageService;
        private void OpenReportDesigner() {
            CustomConnectionProviderService connectionProviderService = new CustomConnectionProviderService();
            connectionStorageService = new CustomConnectionStorageService(connectionProviderService) {
                FileName = "connections.xml",
                IncludeApplicationConnections = false,                
            };
            ReportDesignTool designer = new ReportDesignTool(new XtraReport());
            //designer.DesignRibbonForm.DesignMdiController.DataSourceWizardSettings.SqlWizardSettings.DisableNewConnections = true;            
            ReplaceService(designer.DesignRibbonForm.DesignMdiController, typeof(IConnectionStorageService), connectionStorageService);
            ReplaceService(designer.DesignRibbonForm.DesignMdiController, typeof(IConnectionProviderService), connectionProviderService);

            designer.DesignRibbonForm.DesignMdiController.DesignPanelLoaded += DesignMdiControllerOnDesignPanelLoaded;
            designer.ShowRibbonDesignerDialog();
        }

        private void DesignMdiControllerOnDesignPanelLoaded(object sender, DesignerLoadedEventArgs e) {
            ReplaceService(e.DesignerHost, typeof(IConnectionStorageService), connectionStorageService);
            ReplaceService(e.DesignerHost, typeof(IConnectionProviderService), new CustomConnectionProviderService());
        }
        private void ReplaceService(IServiceContainer container, Type serviceType, object serviceInstance) {
            if (container.GetService(serviceType) != null)
                container.RemoveService(serviceType);
            container.AddService(serviceType, serviceInstance);
        }
    }
}

File: VB/ReportingEUDSaveReportWithoutConnectionParams/Form1.vb
Imports DevExpress.DataAccess.Wizard.Services
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.UserDesigner
Imports System
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Drawing
Imports System.Windows.Forms

Namespace WinEUDSaveReportWithoutConnectionParams

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs)
            OpenReportDesigner()
        End Sub

        Private connectionStorageService As CustomConnectionStorageService

        Private Sub OpenReportDesigner()
            Dim connectionProviderService As CustomConnectionProviderService = New CustomConnectionProviderService()
            connectionStorageService = New CustomConnectionStorageService(connectionProviderService) With {.FileName = "connections.xml", .IncludeApplicationConnections = False}
            Dim designer As ReportDesignTool = New ReportDesignTool(New XtraReport())
            'designer.DesignRibbonForm.DesignMdiController.DataSourceWizardSettings.SqlWizardSettings.DisableNewConnections = true;            
            ReplaceService(designer.DesignRibbonForm.DesignMdiController, GetType(IConnectionStorageService), connectionStorageService)
            ReplaceService(designer.DesignRibbonForm.DesignMdiController, GetType(IConnectionProviderService), connectionProviderService)
            AddHandler designer.DesignRibbonForm.DesignMdiController.DesignPanelLoaded, AddressOf DesignMdiControllerOnDesignPanelLoaded
            designer.ShowRibbonDesignerDialog()
        End Sub

        Private Sub DesignMdiControllerOnDesignPanelLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
            ReplaceService(e.DesignerHost, GetType(IConnectionStorageService), connectionStorageService)
            ReplaceService(e.DesignerHost, GetType(IConnectionProviderService), New CustomConnectionProviderService())
        End Sub

        Private Sub ReplaceService(ByVal container As IServiceContainer, ByVal serviceType As Type, ByVal serviceInstance As Object)
            If container.GetService(serviceType) IsNot Nothing Then container.RemoveService(serviceType)
            container.AddService(serviceType, serviceInstance)
        End Sub
    End Class
End Namespace
